src/GDALUtils.cpp

```diff
@@ -100,6 +100,282 @@ ErrorHandler::~ErrorHandler()
+                                    int32_t& pixel, int32_t& line)
+{
+    pixel = (int32_t)std::floor(inverse[0] + (inverse[1] * x) +
+        (inverse[2] * y));
+    line = (int32_t) std::floor(inverse[3] + (inverse[4] * x) +
+        (inverse[5] * y));
+
+    int xs = m_raster_x_size;
+    int ys = m_raster_y_size;
+
+    if (!xs || !ys)
+        throw pdal_error("Unable to get X or Y size from raster!");
+
+    if (pixel < 0 || line < 0 || pixel >= xs || line  >= ys)
+    {
+        // The x, y is not coincident with this raster
+        return false;
+    }
+
+    return true;
+}
+
+pdal::Dimension::Type::Enum convertGDALtoPDAL(GDALDataType t)
+{
+
+    using namespace pdal::Dimension::Type;
+    switch (t)
+    {
+        case GDT_Byte:
+            return Unsigned8;
+        case GDT_UInt16:
+            return Unsigned16;
+        case GDT_Int16:
+            return Signed16;
+        case GDT_UInt32:
+            return Unsigned32;
+        case GDT_Int32:
+            return Signed32;
+        case GDT_CFloat32:
+            return Float;
+        case GDT_CFloat64:
+            return Double;
+        default:
+            return None;
+    }
+
+    return None;
+}
+
+bool Raster::readBand(std::vector<uint8_t>& data, int nBand)
+{
+    data.resize(m_raster_x_size * m_raster_y_size);
+
+    GDALRasterBandH band = GDALGetRasterBand(m_ds, nBand);
+    if (!band)
+    {
+        std::ostringstream oss;
+        oss << "Unable to get band " << nBand <<
+            " from data source!";
+        throw pdal_error(oss.str());
+    }
+
+    int nXBlockSize(0);
+    int nYBlockSize(0);
+
+    GDALGetBlockSize(band, &nXBlockSize, &nYBlockSize);
+
+    int nXBlocks = (GDALGetRasterBandXSize(band) + nXBlockSize - 1) / nXBlockSize;
+    int nYBlocks = (GDALGetRasterBandYSize(band) + nYBlockSize - 1) / nYBlockSize;
+
+    for (int iYBlock = 0; iYBlock < nYBlocks; iYBlock++)
+    {
+        int nXValid(0); int nYValid(0);
+        for (int iXBlock = 0; iXBlock < nXBlocks; iXBlock++)
+        {
+
+             if ((iXBlock+1) * nXBlockSize > GDALGetRasterBandXSize(band))
+                 nXValid = GDALGetRasterBandXSize(band) - iXBlock * nXBlockSize;
+             else
+                 nXValid = nXBlockSize;
+             if ((iYBlock+1) * nYBlockSize > GDALGetRasterBandYSize(band))
+                 nYValid = GDALGetRasterBandYSize(band) - iYBlock * nYBlockSize;
+             else
+                 nYValid = nYBlockSize;
+
+            int offset = iXBlock * (nXValid * nYValid) + iYBlock * (nXValid * nYValid);
+
+            CPLErr err = GDALReadBlock(band, iXBlock, iYBlock, data.data() + offset);
+            if (err != CPLE_None)
+            {
+                std::ostringstream oss;
+                oss << "unable to read block for ("<<iXBlock <<","<< iYBlock <<")";
+                throw pdal::pdal_error(oss.str());
+            }
+
+        }
+
+    }
+    return true;
```

Comment:
no, but the other read() method does, and I made them both return a bool for consistency.